llvm/lib/Transforms/Scalar/LoopUnrollPass.cpp

```diff
@@ -776,6 +776,11 @@ shouldPragmaUnroll(Loop *L, const PragmaInfo &PInfo,
   if (PInfo.PragmaFullUnroll && TripCount != 0)
     return TripCount;
 
+  // Small MaxTripCount is clearly calculated with "pragma unroll".
```

Comment:
I don't understand what this comment is trying to tell. The code would be clearer without it.

Reply:
ok, let me rm it, I just want to express the "MaxTripCount is clearly calculated" (e.g 17) not roughly estimated （e.g  "int32 I=0, I < UnkowInt32LoopCount, I++", we may roughly set the MaxTripCount is Int_Max (0x7ffffffff)）

Reply:
Ah, I get what you mean now. It would probably be okay to use a higher limit than UnrollMaxUpperBound if pragma is involved, but I agree that simply unrolling to any MaxTripCount would not be safe.